Philips.CodeAnalysis.MsTestAnalyzers/AssertFailAnalyzer.cs

```diff
@@ -0,0 +1,136 @@
+﻿// © 2019 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MsTestAnalyzers
+{
+	internal class AssertMetadata
+	{
+		public ImmutableArray<IMethodSymbol> FailMethods { get; set; }
+	}
+
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class AssertFailAnalyzer : DiagnosticAnalyzer
+	{
+		private const string Title = @"Assert.Fail should not be used if an alternative is more appropriate";
+		private const string MessageFormat = @"Assert.Fail should not be used in this scenario.  Consider using Assert.AreEqual or Assert.IsTrue or Assert.IsNull";
+		private const string Description = @"";
+		private const string Category = Categories.Maintainability;
+
+		private static DiagnosticDescriptor Rule = new DiagnosticDescriptor(Helper.ToDiagnosticId(DiagnosticIds.AssertFail), Title, MessageFormat, Category, DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
```

Comment:
We probably don't want these flags, correct? ie., don't run on generated code, which is presumably not very relevant for an mstest.

Reply:
This is actually consistent with the rest of the mstest analyzers, but yes, we don't care about it in generated code.  Removed.